### ISSUE TITLE
Provide implementations for Mutex

### ISSUE THREAD

**Issue by Strytyp** [djkoloski/rkyv#120]
Hi, is it possible to provide implementations for std::sync::Mutex and std::sync::RwLock?

**Comment by djkoloski** [comment]
I don't think we can provide a structure that gives the same functionality in an archive (since it must be cross-platform), but if you just want a passthrough then that would be a sensible thing to do. A passthrough would essentially just serialize the contained value inline and only provide read access.

**Comment by Strytyp** [comment]
Seems good for me, I will deserialize anyway before using it. I found it useful to be able to use the derive macros in types with Mutex or RwLock directly.

**Comment by djkoloski** [comment]
Linking #122. Once that issue is closed, I'll add a wrapper like `LockMutex` so you can do:

```rust
#[derive(Archive)]
pub struct Example {
    #[with(LockMutex)]
    foo: Mutex<i32>,
}
```

and hopefully it will just work.

**Comment by djkoloski** [comment]
Fixed by bd5ad7fa241c5a1f16e61d275b40ab9c3289933d. This will ship in 0.7.0, but you can use it now by using the 0.7 branch as a git dependency. The syntax for this is:

```rust
use rkyv::with::Lock;

#[derive(Archive)]
pub struct Example {
    #[with(Lock)]
    foo: Mutex<i32>,
}
```

The field will be archived as an `Immutable` which makes it unable to be mutated.
